src/RuleEvaluator.php

```diff
@@ -64,21 +65,37 @@ private static function evaluateCondition(array $subjectAttributes, Condition $c
         if ($value !== null) {
             switch ($condition->operator) {
                 case 'GTE':
-                    return self::compareNumber($value, $condition->value, function($a, $b) {
-                        return $a >= $b;
-                    });
+                    if (is_numeric($value) && is_numeric($condition->value)) {
+                        return self::compareNumber($value, $condition->value, function($a, $b) {
+                            return $a >= $b;
+                        });
```

Comment:
since we now have the `is_numeric()` checks pulled top level, we can ax the use of `compareNumber()` and simplify this a bit.